lib/util/include/hse_util/arch.h

```diff
@@ -136,6 +146,25 @@ get_time_ns(void)
 }
 #endif
 
+/* HSE_L1D_ALIGNED - Used to separate non-related fields in a structure
+ * by one cacheline to avoid false sharing.
+ *
+ * HSE_ACP_ALIGNED - Used to separate non-related fields in a structure
+ * by one or more cachelines to avoid false sharing due to Adjacent
+ * Cacheline Prefetch (e.g., two cache lines on amd64, otherwise one).
+ *
+ * HSE_L1X_ALIGNED - Used to align to the next cacheline when Adjacent
```

Comment:
Greg, not clear on when to use HSE_L1X_ALIGNED. Could you pls. elaborate?

Reply:
Right.  The idea is that sometimes we have a structure with a lock in one cache line and read-mostly fields that immediately follow.  So for amd64 the cache lines are small so we put these fields into separate cache lines to avoid wrecking the read-mostly fields when we acquire the lock, and to prevent someone else trying to get the lock from stealing wrecking our otherwise read-mostly cache line.
But on s390x cache lines are 256-bytes, so in that case I don't want the combined size to be 512-bytes, so L1X does not add an additional cache line alignment in that case.  Make sense?

Reply:
Makes sense now, thanks!